### ISSUE TITLE
getArtifactsLocation() doesn't seem to reflect semantic versioning artifact name

### ISSUE THREAD

**Issue by jguo11** [awslabs/aws-codebuild-jenkins-plugin#77]
I have a CodeBuild project which enables semantic versioning and has namespace type as none. The `getArtifactsLocation()` method incorrectly returns the CodeBuild project's name instead of the artifact name defined in buildspec file (In this case, name = `$(date --iso-8601=ns)`).

My Jenkins pipeline runs the following commands:

```
def codeBuild = awsCodeBuild projectName: 'CodeBuild_foo', \
                                            credentialsType: 'keys', \
                                            accessKey: '', \
                                            awsSecretKey: '', \
                                            region: 'us-west-2', \
                                            sourceControlType: 'project'
print codeBuild.getArtifactsLocation()
```

Expected return value: `arn:aws:s3:::some_paths/2019-02-14T02:32:04,578842365+0000`

Actual return value: `arn:aws:s3:::some_paths/CodeBuild_foo`

The Jenkins task has output:

```
[AWS CodeBuild Plugin] 2019/02/14 02:32:04 Updating artifact name as  2019-02-14T02:32:04,578842365+0000
[AWS CodeBuild Plugin] 2019/02/14 02:32:04 Phase complete: UPLOAD_ARTIFACTS Success: true
[AWS CodeBuild Plugin] 2019/02/14 02:32:04 Phase context status code:  Message: 
[Pipeline] echo
 arn:aws:s3:::some_paths/CodeBuild_foo
```

**Comment by leobaran-aws** [comment]
Thanks for notifying us of this issue. The artifact location is dynamic with semantic versioning and is updated to the overridden name in the middle of the build. The plugin sets the artifact location for the CodeBuildResult at the start of the build when it still has the default value for a blank artifact name (project name). I'm working on a fix now.

**Comment by leobaran-aws** [comment]
Fixed in 95046f99487fee2492168d2afe1545aef0679191, which will be released in 0.35 later today.
